prow/plugins/cherrypickapproved/cherrypickapproved.go

```diff
@@ -0,0 +1,228 @@
+
+}
+
+func (*defaultImpl) RemoveLabel(gc plugins.PluginGitHubClient, org, repo string, number int, label string) error {
+	return gc.RemoveLabel(org, repo, number, label)
+}
+
+func (*defaultImpl) HttpGet(url string) (*http.Response, error) {
+	return http.Get(url)
+}
+
+func (*defaultImpl) CloseBody(resp *http.Response) error {
+	return resp.Body.Close()
+}
+
+func (*defaultImpl) ReadAllBody(resp *http.Response) ([]byte, error) {
+	return io.ReadAll(resp.Body)
+}
+
+func (*defaultImpl) YAMLUnmarshal(y []byte) (map[string]map[string][]string, error) {
+	var result map[string]map[string][]string
+	if err := yaml.Unmarshal(y, &result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func handlePullRequestReviewEvent(pc plugins.Agent, e github.ReviewEvent) error {
+	if err := newHandler().handle(pc.Logger, pc.GitHubClient, e); err != nil {
+		pc.Logger.WithError(err).Error("skipping")
+	}
+	return nil
+}
+
+func (h *handler) handle(log *logrus.Entry, gc plugins.PluginGitHubClient, e github.ReviewEvent) error {
+	funcStart := time.Now()
+	defer func() {
+		log.WithField("duration", time.Since(funcStart).String()).Debug("Completed handlePullRequestReviewEvent")
+	}()
+
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	branch := e.PullRequest.Base.Ref
+
+	if org != defaultKubernetesRepoAndOrg || repo != defaultKubernetesRepoAndOrg || !branchRegexp.MatchString(branch) {
+		return errors.New("no k/k release branch PR review")
+	}
+
+	// Only react to reviews that are being submitted (not editted or dismissed).
+	if e.Action != github.ReviewActionSubmitted {
+		return errors.New("no submit review action")
+	}
+
+	// The review webhook returns state as lowercase, while the review API
+	// returns state as uppercase. Uppercase the value here so it always
+	// matches the constant.
+	if github.ReviewState(strings.ToUpper(string(e.Review.State))) != github.ReviewStateApproved {
+		return errors.New("no approved review state")
+	}
+
+	prNumber := e.PullRequest.Number
+	issueLabels, err := h.GetIssueLabels(gc, org, repo, prNumber)
+	if err != nil {
+		return fmt.Errorf("get issue labels: %w", err)
+	}
+	hasCherryPickApprovedLabel := github.HasLabel(labels.CpApproved, issueLabels)
+	hasCherryPickUnapprovedLabel := github.HasLabel(labels.CpUnapproved, issueLabels)
+	hasLGTMLabel := github.HasLabel(labels.LGTM, issueLabels)
+	hasApprovedLabel := github.HasLabel(labels.Approved, issueLabels)
+	hasInvalidLabels := github.HasLabels(
+		[]string{
+			labels.BlockedPaths,
+			labels.ClaNo,
+			labels.Hold,
+			labels.InvalidOwners,
+			labels.InvalidBug,
+			labels.LGTM,
+			labels.MergeCommits,
+			labels.NeedsOkToTest,
+			labels.NeedsRebase,
+			labels.ReleaseNoteLabelNeeded,
+			labels.WorkInProgress,
+		},
+		issueLabels,
+	)
+
+	isRM, err := h.isReleaseManager(e.Review.User.Login)
+	if err != nil {
+		return fmt.Errorf("check if user is release manager: %w", err)
+	}
+
+	if isRM && hasLGTMLabel && hasApprovedLabel && !hasInvalidLabels {
```

Comment:
Is it possible to check here if all tests are passing? If it is, I think we should add it to the criteria, because we tend to cherry-pick-approve PRs only if tests are passing

Reply:
Yes, I will add that